content/tutorials/trainapp-fitbit.textile

```diff
@@ -0,0 +1,550 @@
+        {props.settingsStorage.getItem("via") && (
+          <TextInput
+            title="Select Station Name"
+            label="Towards"
+            placeholder="Search station"
+            settingsKey="towards" // Used to read the selected value in companion
+            action="Add Item"
+            onAutocomplete={value => {
+              const autoValues = JSON.parse(
+                props.settingsStorage.getItem("via")
+              ).values;
+              return autoValues.filter(option =>
+                option.name.toLowerCase().startsWith(value.toLowerCase())
+              );
+            }}
+          />
+        )}
+      </Section>
+    </Page>
+  );
+}
+
+registerSettingsPage(train);
+```
+You can refer to the Settings guide "here":https://dev.fitbit.com/build/guides/settings/ .
+
+"See this step in Github":https://github.com/ably/tutorials/commit/a1c324a5f28b250ad11973f24298c1acefecd846
+
+h2(#train-data). Step 4 -  Adding static trains data
+
+We'll get started with a few import statements, add the following to your @index.js@ file which is inside the @companion@ folder:
+
+```[javascript]
+import Ably from "ably/browser/static/ably-commonjs.js";
+// Import the messaging module
+import * as messaging from "messaging";
+import { settingsStorage } from "settings";
+import { geolocation } from "geolocation";
+
+let navigator = {userAgent: "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_2) 
+AppleWebKit/537.36 (KHTML, like Gecko) Chrome/79.0.3945.88 Safari/537.36"}
+```
+Then lets add the static data below it from the "gist":https://gist.github.com/vardhanapoorv/b96c5138e7a12b6e6b8f7a02d4d4e76a .
+It contains two arrays, one is used for @StationsIDMapping@ (line & station combination - @stationid@) and other is used for @towardsMapping@ (line & origin combination to get @towards@ options) . 
+
+"See this step in Github":https://github.com/ably/tutorials/commit/66d4a74b1af24ac32b0efebffa23924fb71addcb
+
+h2(#walking-time). Step 5 - Get location & find walking time
+
+Now we will get the location and update the walking time to the user selected station. 
+So for that we will need to use the Google API to get the walking time.
+Follow the steps "here":https://developers.google.com/maps/documentation/distance-matrix/get-api-key , to get the API key.
+Subscribe to the product Ably hub for the TFL London tube schedule "here":https://www.ably.io/hub/products/10 .
+
+The @locationSuccess@ gets triggered everytime the location changes and then we get the new @latitude@ & @longitude@. 
+Now using that location we send it as @origin@ & the @station@ as @destination@ to the Google API.
+Then we get the @walking time@ as response from the Google API, from current location to your selected station. 
+Add the following code in your @index.js@ file which is inside the @companion@ folder: 
+
+```[javascript]
+const ablyAPIKey = '<YOUR-API-KEY>';
+var realtime = new Ably.Realtime(ablyAPIKey);
```

Comment:
Is there a reason we're using `var` for `realtime`, but then `let` for other variables here?